### ISSUE TITLE
Problem with ModuleNotFoundError: No module named 'tensorflow'

### ISSUE THREAD

**Issue by njineermikey1** [MicrosoftDocs/windows-ai-docs#48]
Great writeup.

I'm trying to learn Tensorflow, and I have a Dell machine I can use that has an intel GPU. Following the tutorial (and not being experienced with any of this at all), everything works up to the point I get to the section:

**Install the TensorFlow with DirectML package**

and the lines:

import tensorflow.compat.v1 as tf 

tf.enable_eager_execution(tf.ConfigProto(log_device_placement=True)) 

print(tf.add([1.0, 2.0], [3.0, 4.0])) 

with the error:

SyntaxError: multiple statements found while compiling a single statement

If I only use the first line, I get the error:

Traceback (most recent call last):
  File "<pyshell#0>", line 1, in <module>
    import tensorflow.compat.v1 as t
ModuleNotFoundError: No module named 'tensorflow'

I also tried changing the line:

import tensorflow.compat.v1 as tf

to:

import tensorflow-directml.compat.v1 as tf

because I read that the environment name might be the issue, but I get a syntax error.

So now, I'm stumped. 

Any input?

Mike

BTW, I did find out the Tensorflow install only works when Python is at V3.6. V3.7-V3.10 were all failures.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e11d3a15-5944-5a90-7838-56bdbfff7189
* Version Independent ID: 310912ba-8f79-6db5-55fa-c4fb25fd81f1
* Content: [TensorFlow with DirectML on Windows](https://docs.microsoft.com/en-us/windows/ai/directml/gpu-tensorflow-windows)
* Content Source: [docs/directml/gpu-tensorflow-windows.md](https://github.com/MicrosoftDocs/windows-ai-docs/blob/docs/docs/directml/gpu-tensorflow-windows.md)
* Product: **machine-learning**
* GitHub Login: @stevewhims
* Microsoft Alias: **stwhi**

**Comment by stevewhims** [comment]
Thanks, @njineermikey1, I'm investigating.

-Steve

**Comment by jstoecker** [comment]
Hi,

It looks like the first step failed (running `pip install tensorflow-directml`). Did you get any error messages when running that command? Like you noticed, TFDML only runs in Python 3.5, 3.6, or 3.7: 

![image](https://user-images.githubusercontent.com/3835701/145131102-23fdfec6-e7c6-4abf-87aa-c1856af8cc88.png)

As a side note, you _don't_ want to change the import statement to `import tensorflow-directml` or anything like that. The package may be named tensorflow-directml but the python module is still named `tensorflow`.

**Comment by njineermikey1** [comment]
Thanks guys. New to this software, so bear with me for upcoming obvious questions.

I tried the other packages just as a "maybe this will work" attempt.

I just reran the install and got the same errors as soon as I got to the
python entry part.

Here is everything from the conda command line entry section:


(base) C:\Users\PC_USER>conda create --name directml python=3.6
WARNING: A conda environment already exists at
'C:\Users\PC_USER\.conda\envs\directml'
Remove existing environment (y/[n])? y

Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\PC_USER\.conda\envs\directml

  added / updated specs:
    - python=3.6


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    wheel-0.37.0               |     pyhd3eb1b0_1          33 KB
    ------------------------------------------------------------
                                           Total:          33 KB

The following NEW packages will be INSTALLED:

  certifi            pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3
  pip                pkgs/main/win-64::pip-21.2.2-py36haa95532_0
  python             pkgs/main/win-64::python-3.6.13-h3758d61_0
  setuptools         pkgs/main/win-64::setuptools-58.0.4-py36haa95532_0
  sqlite             pkgs/main/win-64::sqlite-3.36.0-h2bbff1b_0
  vc                 pkgs/main/win-64::vc-14.2-h21ff451_1
  vs2015_runtime     pkgs/main/win-64::vs2015_runtime-14.27.29016-h5e58377_2
  wheel              pkgs/main/noarch::wheel-0.37.0-pyhd3eb1b0_1
  wincertstore       pkgs/main/win-64::wincertstore-0.2-py36h7fe50ca_0


Proceed ([y]/n)? y


Downloading and Extracting Packages
wheel-0.37.0         | 33 KB     |
############################################################################
| 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate directml
#
# To deactivate an active environment, use
#
#     $ conda deactivate


(base) C:\Users\PC_USER>conda activate directml

(directml) C:\Users\PC_USER>pip install tensorflow-directml
Collecting tensorflow-directml
  Using cached tensorflow_directml-1.15.5-cp36-cp36m-win_amd64.whl (69.2 MB)
Requirement already satisfied: wheel>=0.26 in
c:\users\pc_user\.conda\envs\directml\lib\site-packages (from
tensorflow-directml) (0.37.0)
Collecting numpy<1.19.0,>=1.16.0
  Using cached numpy-1.18.5-cp36-cp36m-win_amd64.whl (12.7 MB)
Collecting h5py<=2.10.0
  Using cached h5py-2.10.0-cp36-cp36m-win_amd64.whl (2.4 MB)
Collecting absl-py>=0.7.0
  Using cached absl_py-1.0.0-py3-none-any.whl (126 kB)
Collecting protobuf>=3.6.1
  Using cached protobuf-3.19.1-cp36-cp36m-win_amd64.whl (897 kB)
Collecting opt-einsum>=2.3.2
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting tensorboard<1.16.0,>=1.15.0
  Using cached tensorboard-1.15.0-py3-none-any.whl (3.8 MB)
Collecting gast==0.2.2
  Using cached gast-0.2.2-py3-none-any.whl
Collecting keras-preprocessing>=1.0.5
  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
Collecting google-pasta>=0.1.6
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting grpcio>=1.8.6
  Using cached grpcio-1.42.0-cp36-cp36m-win_amd64.whl (3.2 MB)
Collecting wrapt>=1.11.1
  Using cached wrapt-1.13.3-cp36-cp36m-win_amd64.whl (34 kB)
Collecting keras-applications>=1.0.8
  Using cached Keras_Applications-1.0.8-py3-none-any.whl (50 kB)
Collecting six>=1.10.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting tensorflow-estimator==1.15.1
  Using cached tensorflow_estimator-1.15.1-py2.py3-none-any.whl (503 kB)
Collecting termcolor>=1.1.0
  Using cached termcolor-1.1.0-py3-none-any.whl
Collecting astor>=0.6.0
  Using cached astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting werkzeug>=0.11.15
  Using cached Werkzeug-2.0.2-py3-none-any.whl (288 kB)
Requirement already satisfied: setuptools>=41.0.0 in
c:\users\pc_user\.conda\envs\directml\lib\site-packages (from
tensorboard<1.16.0,>=1.15.0->tensorflow-directml) (58.0.4)
Collecting markdown>=2.6.8
  Using cached Markdown-3.3.6-py3-none-any.whl (97 kB)
Collecting importlib-metadata>=4.4
  Using cached importlib_metadata-4.8.2-py3-none-any.whl (17 kB)
Collecting zipp>=0.5
  Using cached zipp-3.6.0-py3-none-any.whl (5.3 kB)
Collecting typing-extensions>=3.6.4
  Using cached typing_extensions-4.0.1-py3-none-any.whl (22 kB)
Collecting dataclasses
  Using cached dataclasses-0.8-py3-none-any.whl (19 kB)
Installing collected packages: zipp, typing-extensions, six, numpy,
importlib-metadata, dataclasses, werkzeug, protobuf, markdown, h5py,
grpcio, absl-py, wrapt, termcolor, tensorflow-estimator, tensorboard,
opt-einsum, keras-preprocessing, keras-applications, google-pasta, gast,
astor, tensorflow-directml
Successfully installed absl-py-1.0.0 astor-0.8.1 dataclasses-0.8 gast-0.2.2
google-pasta-0.2.0 grpcio-1.42.0 h5py-2.10.0 importlib-metadata-4.8.2
keras-applications-1.0.8 keras-preprocessing-1.1.2 markdown-3.3.6
numpy-1.18.5 opt-einsum-3.3.0 protobuf-3.19.1 six-1.16.0 tensorboard-1.15.0
tensorflow-directml-1.15.5 tensorflow-estimator-1.15.1 termcolor-1.1.0
typing-extensions-4.0.1 werkzeug-2.0.2 wrapt-1.13.3 zipp-3.6.0

(directml) C:\Users\PC_USER>

**Comment by njineermikey1** [comment]
Python Window screenshot

![image](https://user-images.githubusercontent.com/95704652/145222983-e138a259-853a-4717-8df4-e476eca08fb2.png)

**Comment by jstoecker** [comment]
You installed tensorflow-directml correctly in the "directml" conda environment. However, it looks like you're launching a different version of python in the screenshot above. 

It looks to me like you installed Python twice:
1. From https://www.python.org/ (I'm guessing), which you're using to launch the Python 3.6.0 Shell above.
2. Using Miniconda installer, which you then used to create a conda environment named "directml" with Python 3.6.13.

It's totally fine to have multiple python installations, but you need to be careful to use the environment that you expect. From the logs you pasted the conda environment is using python **3.6.13**:

```
  python             pkgs/main/win-64::python-3.6.13-h3758d61_0
```

However, the python shell GUI app is showing python 3.6.0. Another giveaway is that the first line when starting the python interactive shell should show Anaconda somewhere in the name. I would _expect_ to see something like this as the first line if you were using the "directml" environment's python interpreter:

```
Python 3.6.13 |Anaconda, Inc.| (default, Mar 16 2021, 11:37:27) [MSC v.1916 64 bit (AMD64)] on win32
```

In short, you need to make sure you're using the right python environment that has the tensorflow-directml package installed. You can use GUI-based editors and switch environments, but I'm just not familiar with the built-in IDLE/Python shell that comes with CPython installations. You can see some documentation [here](https://code.visualstudio.com/docs/python/environments) for VS Code, which also gives a bit more explanation on python environments. You can also type the program into a .py file using any text editor and run it from the console with the "directml" environment activated.

**Comment by njineermikey1** [comment]
That version (3.6.13) has no files on Python.org. I'll try a few things to see if I can get this sorted out.

**Comment by njineermikey1** [comment]
So, I started python from within the conda command window I was previously running in, and the install/test tutorial worked. Thanks for the help!

**Comment by stevewhims** [comment]
Yes, thanks for the help, @jstoecker! :)

I'll go ahead and close the issue.

Thanks again!
-Steve
